frontend/apps/filemanager/filemanagercollection.lua

```diff
@@ -830,43 +841,89 @@ function FileManagerCollection:onShowCollectionsSearchDialog(search_str)
         parent = search_dialog,
     }
     search_dialog:addWidget(check_button_case)
+    check_button_content = CheckButton:new{
+        text = _("Also search in book content (slow)"),
+        checked = self.include_content,
+        parent = search_dialog,
+    }
+    if self.ui.document then
+        self.include_content = nil
+    else
+        search_dialog:addWidget(check_button_content)
+    end
     UIManager:show(search_dialog)
     search_dialog:onShowKeyboard()
     return true
 end
 
-function FileManagerCollection:searchCollections()
+function FileManagerCollection:searchCollections(coll_name)
     local function isFileMatch(file)
         if self.search_str == "*" then
             return true
         end
         if util.stringSearch(file:gsub(".*/", ""), self.search_str, self.case_sensitive) ~= 0 then
             return true
         end
+        if not DocumentRegistry:hasProvider(file) then
+            return false
+        end
         local book_props = self.ui.bookinfo:getDocProps(file, nil, true) -- do not open the document
+        book_props.display_title = nil
         if next(book_props) ~= nil and self.ui.bookinfo:findInProps(book_props, self.search_str, self.case_sensitive) then
             return true
         end
+        if self.include_content then
+            logger.dbg("Search in book:", file)
+            local ReaderUI = require("apps/reader/readerui")
+            local provider = ReaderUI:extendProvider(file, DocumentRegistry:getProvider(file))
+            local document = DocumentRegistry:openDocument(file, provider)
+            if document then
+                local loaded, found
+                if document.loadDocument then -- CRE
+                    if not self.is_cre_cache_disabled then
```

Comment:
Suggested comment:
```
-- We will be half-loading documents and may mess with crengine's state.
-- Fortunately, this is run in a subprocess, so we won't be affecting the
-- main process's crengine state or any document opened in the main
-- process (we furthermore prevent this feature when one is opened).
-- To avoid creating half-rendered/invalide cache files, it's best to disable
-- crengine saving of such cache files.
```

(Looks like you went back to comment-less coding :) there's not a single one in this PR :/)